test/IntrospectionTests.h

```diff
@@ -0,0 +1,596 @@
+
+        auto FirstAccessor = MakeMemberAccessor("IntVar",&BaseStruct::IntVar);
+        auto SecondAccessor = MakeMemberAccessor("FloatVar",&SingleVarStruct::SetUIntVar,&SingleVarStruct::GetUIntVar);
+        auto ThirdAccessor = MakeMemberAccessor<MemberTags::Local>("VectorVar",&ContainerStruct::GetVectorVar);
+        auto FourthAccessor=MakeMemberAccessor<Combined>("FloatVar",&BaseStruct::SetFloatVar,&BaseStruct::GetFloatVar);
+
+        Boole FirstResult = std::is_same_v<FirstMemberType,decltype(FirstAccessor)>;
+        Boole SecondResult = std::is_same_v<SecondMemberType,decltype(SecondAccessor)>;
+        Boole ThirdResult = std::is_same_v<ThirdMemberType,decltype(ThirdAccessor)>;
+        Boole FourthResult = std::is_same_v<FourthMemberType,decltype(FourthAccessor)>;
+        TEST_EQUAL("MakeMemberAccessor(const_StringView,SetterType,GetterType)-FirstMember",true,FirstResult);
+        TEST_EQUAL("MakeMemberAccessor(const_StringView,SetterType,GetterType)-SecondMember",true,SecondResult);
+        TEST_EQUAL("MakeMemberAccessor(const_StringView,SetterType,GetterType)-ThirdMember",true,ThirdResult);
+        TEST_EQUAL("MakeMemberAccessor(const_StringView,SetterType,GetterType)-FourthMember",true,FourthResult);
+    }// MakeMemberAccessor/BaseStruct
+
+    {// Registration Methods/ContainerStruct
+        TEST_EQUAL("RegisterName()-Dummy",
+                   StringView(""),RegisterName<std::vector<int>>());
+
+        const Boole FirstMatch = std::is_same_v<std::tuple<>,decltype(RegisterMembers<std::vector<int>>())>;
+        TEST_EQUAL("RegisterMembers()-Dummy",
+                   true,FirstMatch);
+
+        TEST_EQUAL("RegisterName()-ContainerStruct",
+                   StringView("ContainerStruct"),RegisterName<ContainerStruct>());
+
+        constexpr MemberTags Local = MemberTags::Local;
+        using ContainerVectorType = decltype(MakeMemberAccessor<Local>("Test",&ContainerStruct::GetVectorVar));
+        using ContainerStringType = decltype(MakeMemberAccessor<Local>("Test",&ContainerStruct::GetStringVar));
+        using ContainerMembersType = std::tuple<ContainerVectorType,ContainerStringType>;
+        const Boole SecondMatch = std::is_same_v<ContainerMembersType,decltype(RegisterMembers<ContainerStruct>())>;
+        TEST_EQUAL("RegisterMembers()-ContainerStruct",
+                   true,SecondMatch);
+    }// Registration Methods/ContainerStruct
+
+    {// Registration Query
+        TEST_EQUAL("IsRegistered()-DerivedStructA",
+                   true,IsRegistered<DerivedStructA>());
+        TEST_EQUAL("IsRegistered()-ContainerStruct",
+                   true,IsRegistered<ContainerStruct>());
+        TEST_EQUAL("IsRegistered()-vector",
+                   false,IsRegistered<std::vector<int>>());
+
+        TEST_EQUAL("GetRegisteredName()-DerivedStructB",
+                   StringView("DerivedStructB"),GetRegisteredName<DerivedStructB>());
```

Comment:
What happens if multiple items of the same type are registered here? Will the same one always be returned?

Reply:
Only if you register a class using the different methods covered in the manual. In which case, yes...it'll always return the same one because it prioritizes GetRegisteredName(), then GetSerializableName(), and RegisterName<>() is last.  So if you have implemented GetRegisteredName() and RegisterName<>(), then RegisterName<>() will always be ignored.  But it will compile.

For the two static methods it'll only search on the class.  A duplicate of the same method will cause a double declaration error.  The template non-member method will also cause a double declaration error if another exists.